olca-app/src/org/openlca/app/editors/projects/reports/model/Report.java

```diff
@@ -25,7 +33,163 @@
   final List<ReportCostResult> addedValues = new ArrayList<>();
   final List<ReportCostResult> netCosts = new ArrayList<>();
 
-  /**
+
+	public static Report fromJsonFile(File reportFile)
+		throws IOException {
+		try (InputStream stream = new FileInputStream(reportFile);
+				 Reader reader = new InputStreamReader(stream, "UTF-8")) {
+			JsonElement json = JsonParser.parseReader(reader);
+			JsonObject obj = json.getAsJsonObject();
+			Report report = new Report();
+
+			var log = LoggerFactory.getLogger(Report.class);
+
+			if (obj.has("title")
+				&& obj.get("title").isJsonPrimitive())
+				report.title = Json.getString(obj, "title");
+			else
+				log.warn("Failed to parse the title of the report.");
+
+			if (obj.has("withNormalisation")
+				&& obj.get("withNormalisation").isJsonPrimitive())
+				report.withNormalisation
+					= Json.getBool(obj, "withNormalisation", false);
+			else
+				log.warn("Failed to parse the withNormalisation value of the report.");
+
+			if (obj.has("withWeighting")
+				&& obj.get("withWeighting").isJsonPrimitive())
+				report.withWeighting = Json.getBool(obj, "withWeighting", false);
+			else
+				log.warn("Failed to parse the withWeighting value of the report.");
```

Comment:
I would love to have a function to avoid repeating this if/else operations, but I do not see how.